src/status_im/contexts/market/view.cljs

```diff
@@ -0,0 +1,69 @@
+(ns status-im.contexts.market.view
+  (:require
+    [quo.context]
+    [quo.core :as quo]
+    [quo.foundations.colors :as colors]
+    [react-native.core :as rn]
+    [status-im.common.home.top-nav.view :as common.top-nav]
+    [status-im.common.refreshable-flat-list.view :as refreshable-flat-list]
+    [status-im.contexts.market.style :as style]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]))
+
+(defn swap-header
+  []
+  (let [theme (quo.context/use-theme)]
+    [rn/view {:style (style/swap-header-container theme)}
+     [quo/text
+      {:weight :semi-bold
+       :size   :heading-1
+       :style  (style/market-header-text theme)}
+      (i18n/label :t/market)]
+     [quo/button
+      {:size      32
+       :icon-left :i/swap
+       :on-press  #()}
+      (i18n/label :t/swap)]]))
+
+(defn sort-header
+  []
+  (let [theme (quo.context/use-theme)]
+    [rn/view {:style (style/sort-header-container theme)}
+     [quo/text
+      {:size  :paragraph-2
+       :style (style/sort-text theme)}
+      (i18n/label :t/market-cap)]
+     [quo/icon :i/arrow-down
+      {:size            12
+       :color           colors/neutral-50
+       :container-style {:margin-left 4}}]]))
+
+(defn token
+  [token-data]
+  [quo/market-token token-data])
+
+(defn view
+  []
+  (let [tokens-loading?                (rf/sub [:wallet/home-tokens-loading?])
+        tokens                         (rf/sub [:market/tokens])
```

Comment:
Assuming the market data will be updating regularly (with some interval), we should avoid/minimise the re-rendering of the entire list and instead re-render only the tokens for which the market data has changed. This can be achieved by passing the flat-list a vector/set of token identifiers instead of entire token maps. Then, each token component can subscribe to the token data by id, leaving the flat-list intact when the data updates. Of course, if the order of the tokens changes, we can't avoid re-rendering the entire list.

Reply:
Thanks for the suggested improvement!
This PR is just a UI part, so it is as simple as possible. 
`status-go` endpoints aren't merged but already accessible in a `status-go` branch so I'm starting to integrate them. I will try to implement this idea in an integration PR to avoid unnecessary re-rendering 👍